astropy/uncertainty/tests/test_distribution.py

```diff
@@ -0,0 +1,305 @@
+                        * np.array([1000, .01, 80, 10])[:, np.newaxis]
+                        + np.array([2000, 0, 0, 500])[:, np.newaxis])
+        # another_data is in pc, but main distr is in kpc
+        another_distr = Distribution(another_data * u.pc)
+        combined_distr = self.distr + another_distr
+
+        expected = np.median(self.data + another_data/1000,
+                             axis=-1) * self.distr.unit
+        assert_quantity_allclose(combined_distr.pdf_median, expected)
+
+        expected = np.var(self.data + another_data/1000, axis=-1) * self.distr.unit**2
+        assert_quantity_allclose(combined_distr.pdf_var, expected)
+
+
+def test_helper_normal_samples():
+    centerq = [1, 5, 30, 400] * u.kpc
+
+    with NumpyRNGContext(12345):
+        n_dist = ds.normal(centerq, std=[0.2, 1.5, 4, 1]*u.kpc, n_samples=100)
+        assert n_dist.distribution.shape == (4, 100)
+        assert n_dist.shape == (4, )
+        assert n_dist.unit == u.kpc
+        assert np.all(n_dist.pdf_std > 100*u.pc)
+
+        n_dist2 = ds.normal(centerq, std=[0.2, 1.5, 4, 1]*u.pc, n_samples=20000)
+        assert n_dist2.distribution.shape == (4, 20000)
+        assert n_dist2.shape == (4, )
+        assert n_dist2.unit == u.kpc
+        assert np.all(n_dist2.pdf_std < 100*u.pc)
+
+
+def test_helper_poisson_samples():
+    centerqadu = [1, 5, 30, 400] * u.adu
+
+    with NumpyRNGContext(12345):
+        p_dist = ds.poisson(centerqadu, n_samples=100)
+        assert p_dist.shape == (4,)
+        assert p_dist.distribution.shape == (4, 100)
+        assert p_dist.unit == u.adu
+        p_min = np.min(p_dist)
+        assert isinstance(p_min, Distribution)
+        assert p_min.shape == ()
+        assert np.all(p_min >= 0)
+        assert np.all(np.abs(p_dist.pdf_mean - centerqadu) < centerqadu)
+
+
+def test_helper_uniform_samples():
+    udist = ds.uniform([1, 2]*u.kpc, [3, 4]*u.kpc)
+    assert udist.shape == (2, )
+    assert udist.distribution.shape == (2, 1000)
+    assert np.all(np.min(udist.distribution, axis=-1) > [1, 2]*u.kpc)
+    assert np.all(np.max(udist.distribution, axis=-1) < [3, 4]*u.kpc)
+
+    # try the alternative creator
+    udist = ds.uniform_center_width([1, 3, 2] * u.pc, [5, 4, 3] * u.pc)
+    assert udist.shape == (3, )
+    assert udist.distribution.shape == (3, 1000)
+    assert np.all(np.min(udist.distribution, axis=-1) > [-1.5, 1, 0.5]*u.pc)
+    assert np.all(np.max(udist.distribution, axis=-1) < [3.5, 5, 3.5]*u.pc)
+
+
+def test_helper_normal_exact():
+    pytest.skip('distribution stretch goal not yet implemented')
+    centerq = [1, 5, 30, 400] * u.kpc
+    ds.normal(centerq, std=[0.2, 1.5, 4, 1]*u.kpc)
+    ds.normal(centerq, var=[0.04, 2.25, 16, 1]*u.kpc**2)
+    ds.normal(centerq, ivar=[25, 0.44444444, 0.625, 1]*u.kpc**-2)
+
+
+def test_helper_poisson_exact():
+    pytest.skip('distribution stretch goal not yet implemented')
+    # TODO: what does Poisson Distribution mean for a quantity in e.g. kpc?
```

Comment:
I think this is easy to implement: just don't allow anything but dimensionless for now - better relax later!